sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,514 @@
+        [R(wavelength, t) = A eff (wavelength, t)G(wavelength)]
+
+
+        Parameters
+        ----------
+        :keyword
+
+        band_width in angstroms - range around channel wavelength that is probed, all other values return as zeros in wavelength response array
+
+        cropped = test case of returning a cropped array
+        Returns
+        -------
+        :return: float, array describing the response per wavelength of effective area (wavelength response)
+                 count rate for spectrum?  : units cm^2 DN phot^-1
+
+        NOTE:
+
+        The energy of a photon E = hf = hc / lambda . Expressing hc in units of eV = 12398.4953
+
+        """
+
+        # conversions
+        dn_per_photon = u.count / u.photon
+
+        var = self.get_channel_data(channel)
+
+        # # move to test
+        # if use_table2_gain:
+        #     self.calculate_effective_area(channel)
+        #     self.calculate_system_gain(channel, use_table2_gain=True)
+        #     self.wavelength_response = self.effective_area * self.system_gain * dn_per_photon
+
+        # # move to test
+        # elif use_response_table2:
+        #     # returns the instrument response values listed in Table 2
+        #     i = self.channel_list.index(channel)
+        #     response = [0.664, 1.785, 3.365, 1.217, 1.14, 0.041, 0.027, 0.0024, 0.0046][i]
+        #     self.wavelength_response = response
+
+        # default is to use these values
+        if use_genx_values:
+
+            self.calculate_system_gain(channel, use_genx_values=True)
+            wavelength_response = var['effective_area'] * (self.system_gain * dn_per_photon)
+
+
+            # if total range removed. float value returned, moving away from this!
+            # else:
+            #     self.wavelength_response = var['effective_area'][channel_index] * (
+            #         self.system_gain * dn_per_photon)
+
+        else:
+            self.calculate_system_gain(channel)
+            self.calculate_effective_area(channel, total_range=True)
+            wavelength_response = self.effective_area * self.system_gain
+
+
+
+            # wavelength range selected, conversion to index by multiplying by ten.
+            # This is because the wavelength array is of dispersed wavelengths.
+            # # + 100 in index = + 10 wavelengths in array
+            # # + 10 in index = + 1 wavelength in array
+        self.range = int(band_width * 10)
+        channel_index = self.wavelength_range_index(channel)
+
+        # mask the wavelength array with zeros except in a small wavelength range near channel
+        for index, value in enumerate(wavelength_response):
+            if index - self.range < channel_index < index + self.range:
+                continue
+            else:
+                wavelength_response[index] = 0.0
+
+        self.wavelength_response = wavelength_response
+
+    def calculate_contribution_function(self, channel, ion_object, chianti_Gofnt=False,
```

Comment:
This should be called `get_contribution_function` or something similar since we are not actually doing the calculation. We are asking ChiantiPy to do it for us and then storing it.